apbs/use-shared-services-across-namespaces.md

```diff
@@ -0,0 +1,54 @@
+# Abstract
+
+APBs for resource intensive services should be programmed to re-use the service that they would otherwise deploy, if that service is already deployed and available in another namespace.
+
+## Terms
+
+- APB ([ansible playbook bundle](https://docs.openshift.org/latest/apb_devel/index.html))
+- Shared Service: A service that is used across multiple namespaces
+- System Resources: CPU or memory
+- Namespace / Project: These refer to the same thing and may be used interchangably.
+- apb.yml: ([APB spec file](https://docs.openshift.org/latest/apb_devel/writing/reference.html#apb-devel-writing-ref-spec)) Configures the inputs passed to an APB prior to deployment.
+
+## Problem Description
+
+Some of the services used in the development of a mobile application can require quite a lot of system resources which could turn into a factor that limits adoption of the Mobile Platform.
+
+For example, if multiple developers are building a product which requires KeyCloak, and they do not have the resources available to run a seperate KeyCloak instance in each developers project, there is currently no convenient way for all of these developers to develop their application using a single instance of KeyCloak.
+
+## Goals
+
+- A convenient way for the developer to tell the platform about an existing deployment of the service he is about to deploy.
+- The APB for a shared service automatically connecting to the shared service in another namespace, using the provided details.
+- The APB for a shared service automatically configuring the shared service in another namespace and exposing the configuration details to the current namespace.
+
+## Non-Goals
+
+- The platform intelligently discovering and connecting to services that are running in seperate namespaces.
+- The platform intelligently discovering and suggesting services to connect to.
+
+## Proposed Solution
+
+For services that are suited to being shared services (such as KeyCloak), we can make a change the apb.yml, allowing the developer to specify connection details for an already existing instance of the service this APB would otherwise deploy.
+
+Example:
+![Image of KeyCloak APB config screen with shared service fields](./images/shared-services-config.png)
+
+In this example, if the `Host of existing KeyCloak service` were left empty then a new service would be installed. However if this field were populated the APB would connect to the host with the provided auth details and create a configuration as required.
```

Comment:
> and create a configuration as required.
and run through the rest of the APB as though the instance had finished provisioning the actual running instance